tiledb/sm/query/reader.h

```diff
@@ -222,8 +216,8 @@ class Reader {
   /** Returns the last fragment uri. */
   URI last_fragment_uri() const;
 
-  /** Initializes the reader. */
-  Status init();
+  /** Initializes the reader with the subarray layout. */
+  Status init(Layout layout);
```

Comment:
let's make this a const-ref to avoid a copy at this layer

Reply:
Will do, though just noting that `Layout` is just a `uint8_t` enum.